### PR TITLE
README rewrite

### PR DESCRIPTION
**Pull request by jsonvillanueva**
Reworded and added more information (particularly about Poetry not being required and entry point/metadata).

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -1,40 +1,49 @@
...
-Plugins are features that can be used to extend the core of Manim's
-functionality. By convention, plugins should be prefixed by ``manim-`` when
-being named. This allows users to easily search for plugins on organizations
-like PyPi, but it is not required. Manim discovers user plugins via the
-metadata specified in the ``pyproject.toml``.
+The only requirement of your preferred build system is that it specifies the
+``manim.plugins`` `entry point
+<https://packaging.python.org/specifications/entry-points/>`_.
+
+.. note::
```
Comment (naveen521kk) [inline README.rst:19]: I don't think github/PyPi supports this syntax?

File: README.rst [hunk 2]
```diff
@@ -89,18 +117,44 @@ This command will read the ``pyproject.toml``, install the dependencies of
...
+to ensure functionality.
+
 Uploading Your Project
 ----------------------
 
-By default, poetry is set to register the package/plugin to pypi. As soon as
-your plugin is useful locally, run the following:
+By default, poetry is set to register the package/plugin to PyPI. You'll need
+to register an account there to upload/update your plugin. As soon as your
+plugin is useful locally, run the following:
```
Comment (naveen521kk) [inline README.rst:157]: Add a note about template repo and also, GitHub action to publish things. They have to set `PyPI` API token in secrets as `PYPI_API_TOKEN` and then just release of Github. Things will be pushed to PyPi automatically.

**Review by leotrs — APPROVED** [review]

File: README.rst (hunk 1 above)
Comment (jsonvillanueva) [reply README.rst:19]: Do you have any such recommendations? (I believe you're referring to the .. note:: syntax). I might convert from .rst to .md, or install a dev-dependency for the package to check syntax against PyPi.

File: README.rst (hunk 2 above)
Comment (jsonvillanueva) [reply README.rst:157]: I'm not sure if this is necessary to document, and I'm also unable to modify the GitHub Action (this requires repo edit permissions). I think you'll need to create project API Token on PyPI and the GitHub Secret so I can document this with a working example.

File: README.rst (hunk 1 above)
Comment (naveen521kk) [reply README.rst:19]: Just remove this `.. note::` like that of sphinx, instead use Headings or somethings. Try to edit in Github Web Editor there is even live preview feature and AFAIK, PyPi supports Github like Syntax.

File: README.rst (hunk 2 above)
Comment (naveen521kk) [reply README.rst:157]: > I think you'll need to create project API Token on PyPI and the GitHub Secret so I can document this with a working example.

I have done it when I have posted this comment. :) 

Just say, you can set you PyPi token in secrets tab, as  `PYPI_API_TOKEN `, and then do a release in Github, and things will be automatically pushed to PyPI.

File: README.rst (hunk 1 above)
Comment (jsonvillanueva) [reply README.rst:19]: It seems like it's supported, but I can still change it. I uploaded to repository.testpypi:
https://test.pypi.org/project/manim-plugintemplate/

Comment (naveen521kk) [reply README.rst:19]: Ok then seems good.

**Review by naveen521kk — APPROVED** [review]
LGTM!

**Comment by jsonvillanueva** [comment]
I added a small bit of information to the repo just after this merge. It's not much but it mentions the Github Actions CI/CD as well as uploading to Test PyPI and the fact that this is a template repository. Should I open another PR? https://github.com/jsonvillanueva/manim-plugintemplate/tree/improve-readme

**Comment by naveen521kk** [comment]
Yeah, open another one.

**Pull request merged**